rust/feature-flags/src/flags/flag_matching.rs

```diff
@@ -1472,9 +1471,25 @@ impl FeatureFlagMatcher {
                         self.process_hash_key_override(hash_key, target_distinct_ids)
                             .await
                     }
-                    // if a flag has experience continuity enabled but no hash key override is provided,
-                    // we don't need to write an override, we can just use the distinct_id
-                    None => (None, false),
+                    // If no hash key override is provided, we need to look up existing overrides
```

Comment:
[nit] if we're gonna write a comment here, IMO it's worth saying the whole thing that you did in the PR description, i.e.

> Most of the time this is fine because posthog-js always includes $anon_distinct_id in its requests, but if a customer is using hash key overrides across posthog-js and a server side sdk then the experience will be inconsistent.

Since that explains why we're doing something that feels non-standard.